multicluster/controllers/multicluster/leader/resourceexport_controller.go

```diff
@@ -461,7 +461,7 @@ func (r *ResourceExportReconciler) updateResourceExportStatus(resExport *mcsv1al
 
 // deleteResourceExport removes ResourceExport finalizer string and updates it, so Kubernetes can complete deletion.
 func (r *ResourceExportReconciler) deleteResourceExport(resExport *mcsv1alpha1.ResourceExport) (ctrl.Result, error) {
-	resExport.SetFinalizers(common.RemoveStringFromSlice(resExport.Finalizers, constants.ResourceExportFinalizer))
+	resExport.SetFinalizers(common.RemoveAllOccurrenceFromSlice(resExport.Finalizers, []string{constants.LegacyResourceExportFinalizer, constants.ResourceExportFinalizer}))
```

Comment:
A similar comment, we could use the following and avoid the new utility function:
```go
finalizers := slices.DeleteFunc(slices.Clone(resExport.Finalizers), func(s string) bool {
        return s == constants.LegacyResourceExportFinalizer || s == constants.ResourceExportFinalizer
})
resExport.SetFinalizers(finalizers)
```

Reply:
Thanks for the suggestion, done